src/bpmn-properties-panel/provider/zeebe/ZeebePropertiesProvider.js

```diff
@@ -92,6 +97,15 @@ function TargetGroup(element) {
   };
 }
 
+function HeaderGroup(element) {
+  return {
+    id: 'headers',
+    label: 'Header',
+    component: ListGroup,
+    ...HeaderProps(element)
+  };
+}
```

Comment:
@barmac the problem of this approach is that the header props are don't accessible for other providers anymore. Currently, when registering a new provider, you will see a structure like that

![image](https://user-images.githubusercontent.com/9433996/120618647-8e18f380-c45b-11eb-81d0-3b983e15207e.png)

If we move the whole generation to a single component using the `ListGroup` under the hood, it would be much harder (maybe impossible) for other providers to access the `items` and manipulate them.

Reply:
Oh, so this is about different props that ListGroup and Group take. While a Group expects just a list of entries, a ListGroup takes also specialised components: add, remove. And then, some props can be rendered components while some have to be functions (add, remove).